mlir/include/mlir/Dialect/MIOpen/LowerMIOpenOps.h

```diff
@@ -702,6 +714,30 @@ computeIndexDiffMap(OpBuilder &b, Location loc,
                 upperIndicesDiff[upperDim]));
       }
 
+      int64_t lowerDim = q[0].template cast<IntegerAttr>().getInt();
+      lowerIndicesDiffMap[lowerDim] = lowerDiff;
+      lowerIndicesUpdatedMap[lowerDim] = b.create<AddIOp>(
+          loc,
+          b.create<IndexCastOp>(loc, lowerIndicesOriginal[lowerDim],
+                                b.getIntegerType(32)),
+          lowerDiff);
+    } else if (transformation.getValue() == "UnMerge2") {
+      auto e = g.get("parameters").template cast<ArrayAttr>();
+      assert(e.size() == p.size());
```

Comment:
it seems the only difference between `UnMerge` and `UnMerge2` is that `UnMerge2` can have more than 2 upper dimensions?

Reply:
yes, UnMerge2 can have more than 2 upper dimensions

Reply:
Can we merge `UnMerge` and `UnMerge2` in such case? The only thing we need to do is to relax the assert in `UnMerge`?

Reply:
the algorithm of UnMerge2 is different current UnMerge